### ISSUE TITLE
Book 3: Improve Monte Carlo integration explanations

### ISSUE THREAD

**Issue by amesgames** [RayTracing/raytracing.github.io#386]
I have a few suggestions for improving clarity of the chapters "One Dimensional MC Integration" and "MC Integration on the Sphere of Directions". I believe this is important to help beginners (such as myself) kickstart their journey into raytracing.

1. In "One Dimensional MC Integration", I think it is crucial to state that the PDF p(x) is equal to the derivative of the CDF P(x). This will help the reader follow how we get from PDF p(r) = r/2 to the piecewise CDF P(X) definition that follows.
1. In "One Dimensional MC Integration", state that the method described in this chapter is called "The Inversion Method" in the literature. This will help readers map the book to other resources.
1. In "MC Integration on the Sphere of Directions", state that the method described is called "The Rejection Method" and can be used when it is not possible to analytically invert the CDF. Otherwise, the reader (myself) might hunt for the CDF. Stating this helps connect that section to the previous one.

This series has really helped me understand the traditional literature and the papers on raytracing. I can help draft some proposed text for these changes, if you all like. Or, just take these as suggestions.

Thanks!

**Comment by trevordblack** [comment]
These are excellent ideas.

There's a balance to strike between being easily digestible and industry standard.
But, since this is "The Rest of Your Life" we can be much more industry standard.

That said, large parts of the text could use some polish.
Don't worry about writing up some text. We'll definitely focus on adding these changes in 3+ when we (I?) edit RestOfYourLife

-------- Original Message --------

**Comment by D-K-E** [comment]
I also second the need for a little more clarity on this issue. I kept wondering how CDF and PDF are related, why are we taking inverses etc. Here is an excerpt from my notes which might help to formulate the relation between PDF, CDF and the matter of inverses. Hope it helps during the editing:


Notice that the CDF of X and PDF of X are related. CDF of X evaluated
for  a sample "k" taken from the domain of X, is simply PDF of X evaluated 
for sample "k" in range `[-\infty, k]`. This makes it possible to define PDF of X using CDF
using the following relation:
```latex
PDF p(k) = Pr[a<x<b] = \int_a^b f_X(k)dx
CDF p(k) = F_X(k) = \int_{-\infty}^k f_X(u)du
f_X = \lim_{h \to 0} \frac{F_X(x+h) - F_X(x)}{h}
```
The nice thing is we can map numbers in range [0, 1] back to the domain of X by taking the inverse of CDF of X and from there we can proceed to PDF of X using the derivative of CDF of X. The mapping is possible because the codomain of CDF of X is in range [0,1].

Or somewhere along those lines.

**Comment by D-K-E** [comment]
On a second note, I would like to add that Foley et. al. 2013, p. 802. provides a great overview on this subject. It might be worth mentioning it. 

I kept wondering why we are not just sampling uniformly the interval of the integral, and after reading the overview, I noticed that the message is simply: given a constant number of samples, if we sample the interval in a non uniform way, we can have a better estimate of the integral. In order to have a better estimate with a uniform distribution it is necessary to increase the number of samples, however with a non uniform distribution that is biased towards samples where the integrand is bigger, we can have a better estimate without needing to increase the number of samples.  

I know that these are not very revelatory remarks, and realize that the discussion about sampling light sources right before the introduction of density functions is practically saying the same thing. I must have lost the trail of thought during the discussion of PDF and CDF.
